### PR TITLE
Framework: Refactor away from _.isArray()

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

Lodash's `isArray` is essentially fully natively supported by `Array.isArray`. This PR replaces the usage, and adds an ESLint rule to warn against using `isArray` from lodash. It might look like a big PR, but replacing one with the other is pretty harmless (neither will throw with any value), so no extensive testing is actually necessary.

If you want to find out why we're moving away from Lodash, see https://github.com/Automattic/wp-calypso/pull/50368 and p4TIVU-9Bf-p2.

#### Testing instructions

* Try importing `isArray` from `lodash` and verify you're getting an ESLint error.
* Verify all tests pass.

#### Notes

There is one ESLint error in one of the files we're touching, and we're intentionally leaving it be. That needs to be refactored away from `create-react-class` (`react/prefer-es6-class`).

### CODE REVIEW

File: client/state/data-layer/wpcom/read/tags/utils.js [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
 
 /**
  * Normalize response from the api so whether we get back a single tag or a list of tags
  * we always pass forward a list
  * Also transform the api response to be something more calypso-friendly
  *
- * @param  {Tag|Tags} apiResponse api response from the tags endpoint
- * @returns {Tags} An object containing all of the normalized tags in the format:
+ * @param  {Object} apiResponse api response from the tags endpoint
+ * @returns {Object} An object containing all of the normalized tags in the format:
```
Comment (sarayourfriend) [inline client/state/data-layer/wpcom/read/tags/utils.js]: Looks like this might should be `Array` at the very least rather than Object (which is just `any`).

**Review by sarayourfriend — APPROVED** [review]
🚀

File: client/state/data-layer/wpcom/read/tags/utils.js (hunk 1 above)
Comment (tyxla) [reply client/state/data-layer/wpcom/read/tags/utils.js]: Good point 👍 Fixed in https://github.com/Automattic/wp-calypso/pull/50582/commits/a57967b6bca048a717cc20dde580954c84926f06

File: client/extensions/woocommerce/state/sites/product-categories/selectors.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /**
  * External dependencies
  */
-import { get, isArray, omit, some, range, values } from 'lodash';
+import { get, omit, some, range, values } from 'lodash';
```
Comment (WunderBart) [inline client/extensions/woocommerce/state/sites/product-categories/selectors.js:4]: Not sure if this isn't already done in another PR, but we could probably also use native [`some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) as well?

Comment (tyxla) [reply client/extensions/woocommerce/state/sites/product-categories/selectors.js:4]: Good call, but yep, that's for another PR 😉  We're addressing those one by one, step by step.
